src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsGrid.component.ts

```diff
@@ -49,18 +57,117 @@ export class NotebookViewsGridComponent implements OnInit {
 
 	ngAfterViewInit() {
 		const self = this;
+		this._activeView = this.activeView;
 
-		self._grid = GridStack.init({
+		this.createGrid();
+
+		this._loaded = true;
+		this.detectChanges();
+
+		self._grid.on('added', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('added', items, self._grid, self._items); } });
+		self._grid.on('removed', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('removed', items, self._grid, self._items); } });
+		self._grid.on('change', function (e: Event, items: GridStackNode[]) { if (self._gridEnabled) { self.persist('change', items, self._grid, self._items); } });
+	}
+
+	ngAfterContentChecked() {
+		if (!this._activeView || this.activeView.guid !== this._activeView.guid) {
+			if (this._grid) {
+				this.destroyGrid();
+				this._grid = undefined;
+			}
+		}
+	}
+
+	ngAfterViewChecked() {
+		if (!this._activeView || this.activeView.guid !== this._activeView.guid) {
```

Comment:
Same here - some comments would be useful to explain what this is supposed to be doing

Reply:
Added some comments to explain